packages/common/src/location/location_strategy.ts

```diff
@@ -81,17 +81,21 @@ export const APP_BASE_HREF = new InjectionToken<string>('appBaseHref');
  * browser's URL.
  *
  * If you're using `PathLocationStrategy`, you must provide a {@link APP_BASE_HREF}
- * or add a base element to the document. This URL prefix that will be preserved
- * when generating and recognizing URLs.
+ * or add a base element to the document.
  *
- * For instance, if you provide an `APP_BASE_HREF` of `'/my/app'` and call
+ * For instance, if you provide an `APP_BASE_HREF` of `'/my/app/'` and call
  * `location.go('/foo')`, the browser's URL will become
- * `example.com/my/app/foo`.
+ * `example.com/my/app/foo`. To ensure all relative URIs resolve correctly,
+ * the base href should end with a `/`.
  *
- * Similarly, if you add `<base href='/my/app'/>` to the document and call
+ * Similarly, if you add `<base href='/my/app/'/>` to the document and call
  * `location.go('/foo')`, the browser's URL will become
  * `example.com/my/app/foo`.
  *
+ * Note that when using `PathLocationStrategy`, neither the query nor
+ * the fragment in the base href will be preserved, as outlined
```

Comment:
```suggestion
 * the fragment in the `<base href>` will be preserved, as outlined
```

or does this cover both `<base href>` and `APP_BASE_HREF`?

Reply:
Only covers `<base href>`. Using a hash and/or fragment in the `APP_BASE_HREF` with `PathLocationStrategy` is incorrect and won't work at all.